src/emitter.js

```diff
@@ -214,13 +214,17 @@ CKEDITOR.define( [ 'eventinfo', 'utils' ], function( EventInfo, utils ) {
 		 * @param {...*} [args] Additional arguments to be passed to the callbacks.
 		 */
 		fire: function( event, args ) {
-			var callbacks = getCallbacksIfAny( this, event );
+			var eventInfo = event instanceof EventInfo && event;
```

Comment:
Hm... I've got no strong opinion about this. Seems that it may be useful, but I'm not 100% sure if passing `EventInfo` object from another event (emitter) will be correct. I mean - it contains a reference to "source" plus some methods. Perhaps it all depends on how we document this (and in the future perhaps other) property. But the `stop()` and `off()` methods create tricky cases to be tested. For instance if we forward one event to another (as we did in bubbling) the `stop()` method can be called in both listeners. Will it cancel both events? Which one it will cancel if it's called in the first listener which ten fires another event using its own `EventInfo` object? :D.

So... a lot to test and a lot to document. If you see a future for this event, then why not, but perhaps it would be better to wait.